src/main.qml

```diff
@@ -517,4 +517,130 @@ ApplicationWindow
             }
         }
     }
+
+    Component.onCompleted: {
+        var query = readCLIArguments(Qt.application.arguments,":");
+        var addr = "";
+
+        if ("parameters" in query) {
+            if ("address" in query.parameters)
+                addr = query.parameters.address;
+        }
+
+        startSearch(addr);
+    }
+
+    function readCLIArguments(argList, sepStr) {
+        // checks argList[] for "-q" = "--query", "--address"
+        // uses an url parser for 'query', but not for 'address'
+        var result = { scheme: "", action: "", comment: "", parameters: {} }
+        var nrArg = argList.length, dStr = "", i = 0, decode = true
+
+        if (sepStr === undefined)
+            sepStr = ":"
+
+        while(i < nrArg-1) {
+            if (argList[i] === "-q" || argList[i] === "--query") {
+                result = schemeComponents(argList[i+1], decode, sepStr)
+                i = nrArg
+            } else if (argList[i] === "--address") {
+                result.parameters.address = argList[i+1]
+                i = nrArg
+            }
+            i++
+        }
+
+        return result
+    }
+
+    function schemeComponents(url, decode, sepStr) {
+        // modified from www.sitepoint.com/get-url-parameters-with-javascript/
```

Comment:
I didn't find any copyright notice from their article, but since you're citing the source, let's say it's fine.

Reply:
I don't know if there is a proper way to do it...